components/canvases/training-canvas.tsx

```
@@ -0,0 +1,247 @@
+  useWorldSimulation(worldRef, camera, dom);
+
+  const { loadFromJson } = useWorldPersistence(worldRef);
+
+  // Load saved world on mount (but don't spawn cars yet)
+
+  // TODO: Update fitness stats and neural network state periodically
+
+  const handleSaveBrain = useCallback(async () => {
+    // TODO: Brain save to local storage
+    toast("Saved Brain", "success");
+  }, [worldRef, generation]);
```

ESLint

[error] Compilation Skipped: Existing memoization could not be preserved

React Compiler has skipped optimizing this component because the existing manual memoization could not be preserved. The inferred dependencies did not match the manually specified dependencies, which could cause the value to change more or less frequently than expected. The inferred dependency was `toast`, but the source dependencies were [worldRef, generation]. Inferred different dependency than source.

/home/jailuser/git/components/canvases/training-canvas.tsx:63:39
  61 |   // TODO: Update fitness stats and neural network state periodically
  62 |
> 63 |   const handleSaveBrain = useCallback(async () => {
     |                                       ^^^^^^^^^^^^^
> 64 |     // TODO: Brain save to local storage
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 65 |     toast("Saved Brain", "success");
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 66 |   }, [worldRef, generation]);
     | ^^^^ Could not preserve existing manual memoization
  67 |
  68 |   const handleExportBrain = useCallback(async () => {
  69 |     // TODO: Brain export logic

(react-hooks/preserve-manual-memoization)